examples/apps/tree-comparison/src/model/legacyTreeInventoryList.ts

```diff
@@ -143,31 +148,32 @@ export class LegacyTreeInventoryList extends DataObject implements IInventoryLis
 			.get<IFluidHandle<LegacySharedTree>>(legacySharedTreeKey)!
 			.get();
 
-		// REV: Not exactly sure why a checkout is required to get "viewChange" events, seems like it should be able
-		// to fire off the legacy shared tree itself?
+		// We must use a checkout in order to get "viewChange" events - it doesn't change any of the rest of our usage though.
 		const checkout = new EagerCheckout(this._tree);
 		// This event handler fires for any change to the tree, so it needs to handle all possibilities (change, add, remove).
 		checkout.on("viewChange", (before: TreeView, after: TreeView) => {
 			const { changed, added, removed } = before.delta(after);
 			for (const quantityNodeId of changed) {
 				const quantityNode = this.tree.currentView.getViewNode(quantityNodeId);
-				// REV: This is annoying to iterate over since I can't filter until I've retrieved the node object.
-				// When adding a node the "inventory" node changes too, but we don't want to handle that here.
+				// When adding/removing an inventory item the "inventory" node changes too, but we don't want to handle that here.
 				if (quantityNode.definition === "quantity") {
 					const newQuantity = quantityNode.payload as number;
 					// eslint-disable-next-line @typescript-eslint/no-non-null-assertion
 					const inventoryItemNodeId = quantityNode.parentage!.parent;
+					const inventoryItemNode =
+						this.tree.currentView.getViewNode(inventoryItemNodeId);
 					// eslint-disable-next-line @typescript-eslint/no-non-null-assertion
-					this._inventoryItems
-						.get(nodeIdToInventoryItemId(inventoryItemNodeId))!
-						.handleQuantityUpdate(newQuantity);
+					const idNodeId = inventoryItemNode.traits.get("id" as TraitLabel)![0];
+					const idNode = this.tree.currentView.getViewNode(idNodeId);
+					const inventoryItemId = idNode.payload as string;
```

Comment:
Why not use the item node's ID directly? You can convert it to UUID form if needed.

Reply:
Ditto for all other usages of this uuid-in-payload. It's not clear to me why this is needed, and appears to be redundant.

Reply:
I actually started with using the NodeId :) I talk about the change in [this comment](https://github.com/microsoft/FluidFramework/pull/17774/files#diff-9a008d41a408f3f612c0f477705db0af80dac145513001e3f7831a5cee07e522R43) - basically in a real-world scenario I expect the user would be explicitly inputting a unique ID (e.g. a UPC code, model number, etc.), not a uuid.

I think a usage principle should probably be "The NodeId is not part of your schema so don't act like it is".

Reply:
Ah, it's like a true item code. Okay, agreed!